contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -506,24 +520,28 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
 
         // Generate the required metadata that we need to append this batch
         uint40 numQueuedTransactions = totalElementsToAppend - numSequencerTransactions;
-        uint40 timestamp;
+        uint40 blockTimestamp;
         uint40 blockNumber;
         if (curContext.numSubsequentQueueTransactions == 0) {
             // The last element is a sequencer tx, therefore pull timestamp and block number from the last context.
-            timestamp = uint40(curContext.timestamp);
+            blockTimestamp = uint40(curContext.timestamp);
             blockNumber = uint40(curContext.blockNumber);
         } else {
             // The last element is a queue tx, therefore pull timestamp and block number from the queue element.
-            Lib_OVMCodec.QueueElement memory lastElement = getQueueElement(nextQueueIndex - 1);
-            timestamp = lastElement.timestamp;
+            Lib_OVMCodec.QueueElement memory lastElement = _getQueueElement(
+                nextQueueIndex - 1,
```

Comment:
can `nextQueueIndex` ever be 0?

Reply:
it's not clear to me why we're subtracting by one here.

Reply:
I had to reread the logic but I'll do my best to explain. If we're inside of this block then `curContext.numSubsequentQueueTransactions > 0` which means that we've processed at least one queue element (or we would've thrown when we tried to process a queue element that doesn't exist). Reason why we want `nextQueueIndex - 1` is because we increment `nextQueueIndex` after processing each queue element. So the index of the last element we processed is `nextQueueIndex - 1`. I can add this as a comment in the code.